components/raftstore/src/store/worker/life.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{Engines, KvEngine, RaftEngine};
+
+/// An assistant to help manage the lifecycle of engines.
+///
+/// Currently, it only enables manual compaction jobs before starting the
+/// engines and disables manual compaction jobs before shutting down the
+/// engines.
+pub struct Assistant<E, R>
```

Comment:
Why construct an asistant struct for doing this rather than using rocksdb directly? Do we have plan to add some other things to manage for asistant?

Reply:
Not yet, as comments for `Assistant`, it's built for further enhancement or developing purpose in the later work.